src/tests/parse_config_file.c

```diff
@@ -986,7 +986,7 @@ int main(int argc, char *argv[])
 	ret = EXIT_SUCCESS;
 non_test_error:
 	(void)unlink(tmpf);
-	c->destroy(c);
+	(void)rmdir(dirname(c->configfile));
```

Comment:
`unlink ` delete the file `lxc-parse-config-file-XXXXXX` in the directory which the testcase runs.
`rmdir` delete the directory `lxc-parse-config-file-XXXXXX` in `lxcpath`, which create by `lxc_contaienr_new`. 

 `c->destroy(c)` will not remove  the directory `lxc-parse-config-file-XXXXXX` in `lxcpath`.